src/main/java/com/code_intelligence/jazzer/mutation/support/RangeSupport.java

```diff
@@ -0,0 +1,151 @@
+    public final float min;
+    public final float max;
+    public final boolean allowNaN;
+
+    public FloatRange(float min, float max, boolean allowNaN) {
+      this.min = min;
+      this.max = max;
+      this.allowNaN = allowNaN;
+    }
+  }
+
+  public static final class DoubleRange {
+    public final double min;
+    public final double max;
+    public final boolean allowNaN;
+
+    public DoubleRange(double min, double max, boolean allowNaN) {
+      this.min = min;
+      this.max = max;
+      this.allowNaN = allowNaN;
+    }
+  }
+
+  public static LongRange resolveIntegralRange(
+      AnnotatedType type, long defaultMin, long defaultMax) {
+    long minValue = defaultMin;
+    long maxValue = defaultMax;
+    for (Annotation annotation : type.getAnnotations()) {
+      if (annotation instanceof InRange) {
+        InRange inRange = (InRange) annotation;
+        if (inRange.min() != Long.MIN_VALUE) {
+          minValue = inRange.min();
+        }
+        if (inRange.max() != Long.MAX_VALUE) {
+          maxValue = inRange.max();
+        }
+      } else if (annotation instanceof Positive) {
+        if (minValue < 1) minValue = 1;
+      } else if (annotation instanceof Negative) {
+        if (maxValue > -1) maxValue = -1;
+      } else if (annotation instanceof NonNegative) {
+        if (minValue < 0) minValue = 0;
+      } else if (annotation instanceof NonPositive) {
+        if (maxValue > 0) maxValue = 0;
+      }
+    }
+    return new LongRange(minValue, maxValue);
+  }
+
+  public static FloatRange resolveFloatRange(
+      AnnotatedType type, float defaultMin, float defaultMax, boolean defaultAllowNaN) {
+    float minValue = defaultMin;
+    float maxValue = defaultMax;
+    boolean allowNaN = defaultAllowNaN;
+    for (Annotation annotation : type.getAnnotations()) {
+      if (annotation instanceof FloatInRange) {
+        FloatInRange floatInRange = (FloatInRange) annotation;
+        minValue = floatInRange.min();
+        maxValue = floatInRange.max();
+        allowNaN = floatInRange.allowNaN();
+      } else if (annotation instanceof Positive) {
+        minValue = Math.max(minValue, Float.MIN_VALUE);
+        allowNaN = false;
+      } else if (annotation instanceof Negative) {
+        maxValue = Math.min(maxValue, -Float.MIN_VALUE);
+        allowNaN = false;
+      } else if (annotation instanceof NonNegative) {
+        minValue = Math.max(minValue, 0.0f);
+        allowNaN = false;
+      } else if (annotation instanceof NonPositive) {
+        maxValue = Math.min(maxValue, 0.0f);
+        allowNaN = false;
+      }
+    }
+    return new FloatRange(minValue, maxValue, allowNaN);
+  }
+
+  public static DoubleRange resolveDoubleRange(
+      AnnotatedType type, double defaultMin, double defaultMax, boolean defaultAllowNaN) {
+    double minValue = defaultMin;
+    double maxValue = defaultMax;
+    boolean allowNaN = defaultAllowNaN;
+    for (Annotation annotation : type.getAnnotations()) {
```

Comment:
In this for-loop (and the equivalents for floats and integrals), every annotation gets to move min/maxValue/allowNan. So if you have something like `@InRange(min=-10, max=10) @Positive int n`, n will be in the merged range [1; 10].
I think, each annotation should set /all/ variables associated with the range. In this way only the last annotation (closest to the type) will be actually active.